framework/wazuh/core/indexer/bulk.py

```diff
@@ -0,0 +1,199 @@
+from typing import Optional, Dict, Any, List, Protocol, Coroutine
+from enum import Enum
+
+from opensearchpy import AsyncOpenSearch
+
+
+class BulkAction(str, Enum):
+    """Enum representing bulk actions for OpenSearch.
+
+    Attributes
+    ----------
+    CREATE : str
+        Action to create a new document.
+    CREATE_OR_UPDATE : str
+        Action to create a new document or update an existing one.
+    UPDATE : str
+        Action to update an existing document.
+    DELETE : str
+        Action to delete an existing document.
+    """
+    CREATE = "create"
+    CREATE_OR_UPDATE = "index"
+    UPDATE = "update"
+    DELETE = "delete"
+
+    @classmethod
+    def exists(cls, action: str) -> bool:
+        try:
+            cls(action)
+        except ValueError:
+            return False
+        return True
+
+
+class BulkMetadata:
+    """Class to handle metadata for a bulk action in OpenSearch.
```

Comment:
```suggestion
    """Class to handle metadata for a bulk action in the Indexer.
```

Reply:
Changed in f3067e1c80a1254db438f9ef7b4bc8535cd89e81.